src/dmd/escape.d

```diff
@@ -835,25 +834,24 @@ ByRef:
             if (!(va && va.isScope()))
                 notMaybeScope(v);
 
-            if ((v.storage_class & (STC.ref_ | STC.out_ | STC.scope_)) && p == sc.func)
+            if (!((v.storage_class & (STC.ref_ | STC.out_ | STC.scope_)) && p == sc.func))
```

Comment:
```suggestion
            if ((v.storage_class & (STC.ref_ | STC.out_ | STC.scope_)) || p != sc.func)
```

Reply:
Ahh that was a misplaced `)` the whole expression should be negated.